Sources/OAuth2Swift.swift

```diff
@@ -118,32 +124,24 @@ open class OAuth2Swift: OAuthSwift {
             }
         }
 
-        var queryString = "client_id=\(self.consumerKey)"
-        if encodeCallbackURL {
-            queryString += "&redirect_uri=\(callbackURL.absoluteString.urlEncodedString)"
-        } else {
-            queryString += "&redirect_uri=\(callbackURL.absoluteString)"
-        }
-        queryString += "&response_type=\(self.responseType)"
-        if !scope.isEmpty {
-            queryString += "&scope=\(scope)"
-        }
-        if !state.isEmpty {
-            queryString += "&state=\(state)"
-        }
-        for param in parameters {
-            queryString += "&\(param.0)=\(param.1)"
-        }
+        var queryString: String? = ""
 
-        var urlString = self.authorizeUrl
-        urlString += (self.authorizeUrl.contains("?") ? "&" : "?")
+        queryString = queryString?.urlQueryByAppending(parameter: "client_id", value: self.consumerKey)
+        queryString = queryString?.urlQueryByAppending(parameter: "redirect_uri", value: self.encodeCallbackURL ? callbackURL.absoluteString.urlEncoded : callbackURL.absoluteString, encode: self.encodeCallbackURLQuery)
+        queryString = queryString?.urlQueryByAppending(parameter: "response_type", value: self.responseType)
+        queryString = queryString?.urlQueryByAppending(parameter: "scope", value: scope)
```

Comment:
That's fine, they are [checked in `urlQueryByAppending` and not added if empty](https://github.com/ianbytchek/OAuthSwift/blob/3915fcace18ede9bf21d7767a16ac80df967db96/Sources/String%2BOAuthSwift.swift#L30).